### PR TITLE
REFACT: mesh region settings

### PR DESCRIPTION
**Pull request by lorenzovecchietti**

### CODE REVIEW

**Comment by SMoraisAnsys** [comment]
Following our discussion I took the liberty to put this PR in draft mode to avoid early merge.
Feel free to contact me if you require help for the changes we did talk about.

**Comment by lorenzovecchietti** [comment]
> Following our discussion I took the liberty to put this PR in draft mode to avoid early merge. Feel free to contact me if you require help for the changes we did talk about.

I think you wanted to do this on #4580, right? :)

File: pyaedt/modules/MeshIcepak.py [hunk 1]
```diff
@@ -572,13 +573,13 @@ def __delitem__(self, key):
         self._app.logger.error("Setting cannot be removed.")
 
     def __iter__(self):
-        return self._instance_settings.__iter__()
+        return self.keys().__iter__()
```
Comment (SMoraisAnsys) [inline pyaedt/modules/MeshIcepak.py:576]:
```suggestion
        return iter(self.keys())
```

File: pyaedt/modules/MeshIcepak.py [hunk 2]
```diff
@@ -572,13 +573,13 @@ def __delitem__(self, key):
...
 
     def __len__(self):
-        return self._instance_settings.__len__()
+        return self.keys().__len__()
```
Comment (SMoraisAnsys) [inline pyaedt/modules/MeshIcepak.py:579]:
```suggestion
        return len(self.keys())
```

File: pyaedt/modules/MeshIcepak.py [hunk 3]
```diff
@@ -572,13 +573,13 @@ def __delitem__(self, key):
...
 
     def __contains__(self, x):
-        return self._instance_settings.__contains__(x)
+        return self.keys().__contains__(x)
```
Comment (SMoraisAnsys) [inline pyaedt/modules/MeshIcepak.py:582]:
```suggestion
        return x in self.keys()
```

**Review by SMoraisAnsys — APPROVED** [review]
LGTM
Unless necessary, I prefer to avoid calling python magic methods through their name `__NAME__` so I left changes proposition.

**Comment by SMoraisAnsys** [comment]
> > Following our discussion I took the liberty to put this PR in draft mode to avoid early merge. Feel free to contact me if you require help for the changes we did talk about.
> 
> I think you wanted to do this on #4580, right? :)

Yes totally, right. Sry for that !

**Comment by MaxJPRey** [comment]
> LGTM Unless necessary, I prefer to avoid calling python magic methods through their name `__NAME__` so I left changes proposition.

Agreed.

**Pull request merged**